x-pack/heartbeat/monitors/browser/project.go

```diff
@@ -167,3 +189,63 @@ func (p *Project) plugin() plugin.Plugin {
 		Endpoints: 1,
 	}
 }
+
+type argsIterator struct {
+	i    int
+	args []string
+	val  string
+}
+
+func (a *argsIterator) Next() bool {
+	if a.i >= len(a.args) {
+		return false
+	}
+	a.val = a.args[a.i]
+	a.i++
+	return true
+}
+
+func (a *argsIterator) Val() string {
+	return a.val
+}
+
+func (a *argsIterator) Peek() (val string, ok bool) {
+	if a.i >= len(a.args) {
+		return "", false
+	}
+
+	val = a.args[a.i]
+	ok = true
+
+	return val, ok
+}
+
+// Iterate through list and filter dev flags + potential params
+func filterDevFlags(args []string, filter map[string]int) []string {
+	result := []string{}
+
+	iter := argsIterator{i: 0, args: args}
+	for {
+		next := iter.Next()
+
+		if !next {
+			break
+		}
+
+		if pCount, ok := filter[iter.Val()]; ok {
+		ParamsIter:
+			for i := 0; i < pCount; i++ {
+				// Found filtered flag, check if it has associated params
+				if param, ok := iter.Peek(); ok && !strings.HasPrefix(param, "-") {
```

Comment:
we have a variadic options for some of the params like `tags` which can spawn multiple values, We would need to handle that as well. Also add a test case.

I would also drop those options from the UI origin as we don't want any level of filtering from the UI for project based monitors - https://github.com/elastic/synthetics/issues/780